lib/apiservers/portlayer/restapi/handlers/containers_handlers.go

```diff
@@ -223,6 +224,9 @@ func (handler *ContainersHandlersImpl) RemoveContainerHandler(params containers.
 
 	err := h.Container.Remove(context.Background(), handler.handlerCtx.Session)
 	if err != nil {
+		if strings.Contains(err.Error(), "The attempted operation cannot be performed in the current state (Powered on)") {
```

Comment:
You should instead return a typed error instead. See https://blog.golang.org/error-handling-and-go